### PR TITLE
docs: unify module pages

### PR DESCRIPTION
**Pull request by thisisjofrank**
This is a looooong one, and doesn't cover private repositories (which I guess it could, but that feels like maybe it should go under reference guides rather than fundamentals?)

### CODE REVIEW

File: runtime/fundamentals/modules.md [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+  - /runtime/manual/advanced/publishing/
+  - /runtime/manual/basics/vendoring/
+  - /runtime/manual/node/cdns.md
+  - /runtime/manual/advanced/http_imports/
+---
+
+Deno standardizes the way modules are imported in both JavaScript and TypeScript
+using the
+[ECMAScript module standard](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules)
+by default. In line with this standard, **file names must be specified in
```
Comment (ry) [inline runtime/fundamentals/modules.md:20]: "Deno standardizes the way modules are imported..." is odd phrasing

How about "Deno uses ECMAScript modules exclusively. That is, `import` rather than `require`."

Maybe add a bit of context about why -- something like "since 2015(?) ES Modules have been part of the specification that defines JavaScript. As such ES Modules work in the browser, CommonJS does not. Deno's goal is to narrow the gap between browser and server."

File: runtime/fundamentals/modules.md [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+full**. You should not omit the file extension and there is no special handling
+of `index.js`.
```
Comment (ry) [inline runtime/fundamentals/modules.md:22]: This belongs in its own section - not part of the introduction. Also this is specific to local modules - it does not apply to https modules, npm modules, or jsr modules.

Something like:

> ## Local imports
> 
> To link to code in your project, use relative imports. These always start with "./" or "../" and end in a file extension. For example... 

After this section about local imports then add a section about remote imports aka third party modules - third party modules always start like this: `https://` or `npm:` or `jsr:`

Once these are explained, then talk about bare specifiers and import maps. Followed by more detailed information.

File: runtime/fundamentals/modules.md [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+It is possible to specify a version range for the package you are importing.
+This is done using the `@` symbol followed by a version range specifier, and
+follows the [semver](https://semver.org/) versioning scheme.
+
+```bash
+deno add @scopename/mypackage           # latest version
+deno add @scopename/mypackage@16.1.0    # exact version
+deno add @scopename/mypackage@^16.1.0   # latest patch version 16.x
+deno add @scopename/mypackage@~16.1.0   # latest version that doesn't increment the first non-zero portion
+```
```
Comment (ry) [inline runtime/fundamentals/modules.md:197]: It's not just with `deno add`, also within specifiers.

The manual should explain the syntax of version ranges https://gist.github.com/jonlabelle/706b28d50ba75bf81d40782aa3c84b3e

File: runtime/fundamentals/modules.md [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+Deno is able to handle network requests through a proxy server, useful for
+various reasons such as security, caching, or accessing resources behind a
+firewall. The runtime supports supports proxies for module downloads and the Web
+standard `fetch` API.
+
+Deno reads proxy configuration from environment variables: `HTTP_PROXY`,
+`HTTPS_PROXY` and `NO_PROXY`.
+
+In case of Windows, if environment variables are not found, Deno falls back to
+reading proxies from the registry.
```
Comment (ry) [inline runtime/fundamentals/modules.md:277]: This seems out of place. The proxy configuration works for modules but also for `fetch`. 

Maybe it makes sense here if its also written in the env var reference docs.

File: runtime/fundamentals/modules.md [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+You can make your Deno modules available to Node.js users with the
+[dnt](https://github.com/denoland/dnt) build tool.
+
+dnt allows you to develop your Deno module mostly as-is and use a single Deno
+script to build, type check, and test an npm package in an output directory.
+Once built, you only need to `npm publish` the output directory to distribute it
+to Node.js users.
+
+For more details, see
+[https://github.com/denoland/dnt](https://github.com/denoland/dnt).
```
Comment (ry) [inline runtime/fundamentals/modules.md:299]: Modules can be published either to JSR or NPM or accessed via HTTPS (deno.land/x being one option). This needs to be reworked.

File: runtime/fundamentals/modules.md [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+You can force deno to refetch and recompile modules into the cache using the
+`--reload` flag of the `deno cache` or `deno run` subcommand.
+
+```bash
+# Reload everything
+deno cache --reload my_module.
+
+# Reload a specific module
+deno cache --reload=jsr:@std/fs my_module.ts
+```
```
Comment (ry) [inline runtime/fundamentals/modules.md:315]: Maybe call this section "Flushing cached dependencies" 

the examples are better if they use `deno run`

File: runtime/getting_started/first_project.md [hunk 7]
```diff
@@ -1,69 +0,0 @@
----
-title: Making a Deno project
-oldUrl: /runtime/manual/getting_started/first_steps/
----
```
Comment (ry) [inline runtime/getting_started/first_project.md:4]: The removal of this file seems like an unrelated change?

File: runtime/manual/basics/modules/index.md [hunk 8]
```diff
@@ -1,237 +0,0 @@
...
-```
-
-### Supported package registries and specifiers
-
-We support the following registries:
-
-| Registry | Specifier |
-| -------- | --------- |
-| JSR      | jsr:      |
-| npm      | npm:      |
```
Comment (ry) [inline runtime/manual/basics/modules/index.md:150]: This information is important and seems excluded

**Review by ry — COMMENTED** [review]
This needs more work before merging. It would be helpful to discuss this with David, Luca, and/or Bartek.

File: runtime/manual/basics/modules/index.md (hunk 8 above)
Comment (thisisjofrank) [reply runtime/manual/basics/modules/index.md:150]: its in there, line 102/

**Review by marvinhagemeister — APPROVED** [review]

File: runtime/fundamentals/modules.md [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+---
+
+Deno uses
+[ECMAScript modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules)
+as its default module system to align with modern JavaScript standards and to
+promote a more efficient and consistent development experience. It's the
+official standard for JavaScript modules, allows for better tree-shaking,
+improved tooling integration, and native support across different environments.
+
+By adopting ECMAScript modules, Deno ensures compatibility with the evolving
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:27]: This sentence is loooooong and a bit unwieldy, I'd suggest splitting it up a bit.

By adopting ECMAScript modules, Deno ensures compatibility with the ever-evolving JavaScript ecosystem. For developers, this means a streamlined and predictable module system that avoids the complexities associated with legacy module formats like CommonJS.

File: runtime/fundamentals/modules.md [hunk 10]
```diff
@@ -0,0 +1,397 @@
...
+// imports the `calc.ts` module next to this file
+import { add } from "./calc.ts";
+
+console.log(add(1, 2)); // 3
+```
+
+You can run this example by calling `deno run main.ts` in the directory that
+contains both `main.ts` and `calc.ts`.
+
+With ECMAScript modules, local import specifiers must always included the full
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:52]:
```suggestion
With ECMAScript modules, local import specifiers must always include the full
```

File: runtime/fundamentals/modules.md (hunk 9 above)
Comment (marvinhagemeister) [reply runtime/fundamentals/modules.md]: Good point, that reads much more natural. Looks like the German came through in me!

File: runtime/fundamentals/modules.md [hunk 11]
```diff
@@ -0,0 +1,397 @@
...
+// WRONG: missing file extension
+import { add } from "./calc";
+
+// CORRECT: includes file extension
+import { add } from "./calc.ts";
+```
+
+## Importing third party modules and libraries
+
+Using third modules in Deno uses the same `import` syntax as your local code.
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:65]:
```suggestion
When working with third-party modules in Deno, use the same `import` syntax as you do for local code.
```

File: runtime/fundamentals/modules.md [hunk 12]
```diff
@@ -0,0 +1,397 @@
...
+`jsr:` , `npm:` or `https://` for URL imports.
+
+```ts title="main.ts"
+import { camelCase } from "jsr:@luca/cases@1.0.0";
+import { say } from "npm:cowsay@1.6.0";
+import { pascalCase } from "https://deno.land/x/case/mod.ts";
+```
+
+You'll find most Deno-related code on [JSR](https://jsr.io), which is the
+recommended registry to use with Deno.
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:76]:
```suggestion
Deno recommends [JSR](https://jsr.io), the modern JavaScript registry, for third party modules. There, you'll find plenty of well documented ES modules for your projects, including the [Deno Standard Library](/runtime/fundamentals/standard_library/).
```

File: runtime/fundamentals/modules.md [hunk 13]
```diff
@@ -0,0 +1,397 @@
...
+
+## Managing third party modules and libraries
+
+Typing out the module with the full version specifier can become tedious real
+fast when you're importing them from multiple files. We can centralize the
+management of remote modules via the `imports` key in `deno.json`. The `imports`
+key is used to remap specifiers. We call this the import map.
+
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:84]:
```suggestion
Typing out the module name with the full version specifier can become tedious when importing them in multiple files. You can centralize management of remote modules with an `imports` field in your `deno.json` file. We call this `imports` field the **import map**.

```

File: runtime/fundamentals/modules.md [hunk 14]
```diff
@@ -0,0 +1,397 @@
...
+{
+  "imports": {
+    "@luca/cases": "jsr:@luca/cases@^1.0.0",
+    "cowsay": "npm:cowsay@^1.6.0",
+    "cases": "https://deno.land/x/case/mod.ts"
+  }
+}
+```
+
+With the remapped specifiers our code gets a bit simpler:
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:95]:
```suggestion
With remapped specifiers, the code looks cleaner:
```

File: runtime/fundamentals/modules.md [hunk 15]
```diff
@@ -0,0 +1,397 @@
...
+
+The remapped name can be any valid specifier. It's a very powerful feature in
+Deno that can remap anything. Learn more about everything the import map can do
+[here](/runtime/manual/basics/import_maps/).
+
+## Adding dependencies with deno add
+
+The installation process is made easier with the `deno add` subcommand. It will
+automatically add the latest version of the package you requested to the
+`imports` section in `deno.json`.
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:107]:
```suggestion
The installation process is made easy with the `deno add` subcommand. It will
automatically add the latest version of the package you requested to the
`imports` section in `deno.json`.
```

File: runtime/fundamentals/modules.md [hunk 16]
```diff
@@ -0,0 +1,397 @@
...
+
+- [deno.land/x](https://deno.land/x)
+- [esm.sh](https://esm.sh)
+- [unpkg.com](https://unpkg.com)
+
+URL imports are useful if you have a small, often single file, Deno project that
+doesn't require any other configuration because you can avoid having a
+`Deno.json` file at all. It's not advised to use this style of import in larger
+applications as you may end up with version conflicts (where different files use
+different version specifiers).
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:190]:
```suggestion
URL imports are useful if you have a small, often single file, Deno project that
doesn't require any other configuration. With URL imports, you can avoid having a
`Deno.json` file at all. It is **not** advised to use this style of import in larger
applications however, as you may end up with version conflicts (where different files use different version specifiers).
```

File: runtime/fundamentals/modules.md [hunk 17]
```diff
@@ -0,0 +1,397 @@
...
+best experience.**
+
+:::
+
+### Overriding URL imports
+
+The other situation where import maps can be very useful is to override URL
+imports in specific modules.
+
+Let's say you want to override a `https://deno.land/x/my-library@1.0.0/mod.ts`
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:207]: can we use a jsr example rather than deno.land?

File: runtime/fundamentals/modules.md [hunk 18]
```diff
@@ -0,0 +1,397 @@
...
+  "https://deno.land/std@0.224.0/io/util.ts": "ae133d310a0fdcf298cea7bc09a599c49acb616d34e148e263bcb02976f80dee",
+  "https://deno.land/std@0.224.0/async/delay.ts": "35957d585a6e3dd87706858fb1d6b551cb278271b03f52c5a2cb70e65e00c26a",
+   ...
+}
+```
+
+### Auto-generated lockfile
+
+As mentioned above, when a Deno configuration file is resolved (ex. `deno.json`)
+then an additive lockfile will be automatically generated. By default, the path
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:330]:
```suggestion
then a lockfile will be automatically generated. By default, the path
```

File: runtime/fundamentals/modules.md [hunk 19]
```diff
@@ -0,0 +1,397 @@
...
+of this lockfile will be `deno.lock`. You can change this path by updating your
+`deno.json` to specify this:
+
+```json title="deno.json"
+{
+  "lock": "./lock.file"
+}
+```
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:338]:
```suggestion
As mentioned above, when a Deno configuration file is resolved (eg. `deno.json`), a lockfile will be automatically generated. By default, the path
of this lockfile will be in the directory root - `deno.lock`. You can change this path by updating your
`deno.json` to specify this:

```json title="deno.json"
{
  "lock": "./lock.file"
}
```
```

File: runtime/fundamentals/modules.md [hunk 20]
```diff
@@ -0,0 +1,397 @@
...
+deno test --allow-read src
+```
+
+### Runtime verification
+
+Like caching above, you can also use lock files during use of the `deno run` sub
+command, validating the integrity of any locked modules during the run. Remember
+that this only validates against dependencies previously added to the lock file.
+
+You can take this a step further as well by using the `--cached-only` flag to
```
Comment (thisisjofrank) [inline runtime/fundamentals/modules.md:385]:
```suggestion
You can take this a step further by using the `--cached-only` flag to
```

File: runtime/manual/tools/env_variables.md [hunk 21]
```diff
@@ -84,10 +83,22 @@ permission to read the environment variables by checking the value of
 ### NO_PROXY
 
 Indicates hosts which should bypass the proxy set in the other environment
-variables. See the [Proxies](../basics/modules/proxies.md) section for more
-information.
+variables. See the [Proxies](#proxiesd) section for more information.
```
Comment (thisisjofrank) [inline runtime/manual/tools/env_variables.md:86]: typo?
```suggestion
variables. See the [Proxies](#proxies) section for more information.
```

File: runtime/manual/tools/env_variables.md [hunk 22]
```diff
@@ -84,10 +83,22 @@ permission to read the environment variables by checking the value of
...
+
+Deno is able to handle network requests through a proxy server, useful for
+various reasons such as security, caching, or accessing resources behind a
+firewall. The runtime supports supports proxies for module downloads and the Web
+standard `fetch` API.
+
+Deno reads proxy configuration from environment variables: `HTTP_PROXY`,
+`HTTPS_PROXY` and `NO_PROXY`.
+
+In case of Windows, if environment variables are not found, Deno falls back to
```
Comment (thisisjofrank) [inline runtime/manual/tools/env_variables.md:103]:
```suggestion
On Windows, if environment variables are not found, Deno falls back to
```

File: runtime/fundamentals/modules.md (hunk 17 above)
Comment (marvinhagemeister) [reply runtime/fundamentals/modules.md:207]: URL imports cannot be overridden with the `scopes` import feature afaik
